### PR TITLE
update amazon-lambda-archetype to provide SAM templates

### LINKED ISSUE #5037: Amazon Lambda SAM CLI integration
**Description**
I've tested the SAM integration that is on the Amazon Lambda HTTP docs, and used a similar sam.jvm.yaml (without the dynamo db YAML), and it works fine.

Any chance the archetype and documentation could clone this functionality from the Amazon Lambda HTTP implementation (incl. uber jar) -- particularly whilst it's fresh in the minds of the developers.

```
./extensions/amazon-lambda-http/maven-archetype/target/classes/archetype-resources/sam.jvm.yaml
./extensions/amazon-lambda-http/maven-archetype/target/classes/archetype-resources/sam.native.yaml
./extensions/amazon-lambda-http/maven-archetype/src/main/resources/archetype-resources/sam.jvm.yaml
./extensions/amazon-lambda-http/maven-archetype/src/main/resources/archetype-resources/sam.native.yaml

```

```
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31

Globals:
  Function:
    Timeout: 5
Resources:
  QuarkusSam:
    Type: AWS::Serverless::Function
    Properties:
      Handler: io.quarkus.amazon.lambda.runtime.QuarkusStreamHandler::handleRequest
      Runtime: java8
      CodeUri: target/lambda-kotlin-1.0-SNAPSHOT-runner.jar
      MemorySize: 128
      Policies: AWSLambdaBasicExecutionRole

```

It would be super cool also to have this in Gradle which is my preferred build tool.

### PR DESCRIPTION
**Pull request by evanchooly**
merge helper scripts to one file and parameterize it to reduce clutter and repetition

fixes #5037

### CODE REVIEW

**Review by gastaldi — COMMENTED** [review]
That looks good to me, but I'll wait until someone with more experience in AWS to approve ;)

**Comment by geoand** [comment]
I am grossly underqualified to review this since I don't know much of anything about amazon lambda :)

**Comment by evanchooly** [comment]
it's mostly just the textual changes (not strictly amazon-related) that I wanted a newbie's eye on.  And the two of you reviewed the last round so i figured I'd give it another shot.  @oztimpower has verified the template changes, fwiw.

**Comment by geoand** [comment]
Sure, I'll go over the text, no problem

File: docs/src/main/asciidoc/amazon-lambda.adoc
Comment (geoand) [inline docs/src/main/asciidoc/amazon-lambda.adoc]: `you choose to` -> `you choose to use` perhaps?

File: docs/src/main/asciidoc/amazon-lambda.adoc
Comment (geoand) [inline docs/src/main/asciidoc/amazon-lambda.adoc]: `return output` maybe better as just plain `output`?

File: docs/src/main/asciidoc/amazon-lambda.adoc
Comment (geoand) [inline docs/src/main/asciidoc/amazon-lambda.adoc]: `as needed` seems redundant

File: docs/src/main/asciidoc/amazon-lambda.adoc
Comment (geoand) [inline docs/src/main/asciidoc/amazon-lambda.adoc]: This sentence doesn't seem right

Comment (evanchooly) [reply docs/src/main/asciidoc/amazon-lambda.adoc]: updated s/executable/image/.  better?

Comment (geoand) [reply docs/src/main/asciidoc/amazon-lambda.adoc]: Now I see what I think is missing: `tested the sam.native.yaml` I assume should be `tested using the sam.native.yaml`

Comment (gsmet) [reply docs/src/main/asciidoc/amazon-lambda.adoc]: FTR, `native executable` is the wording we agreed on.

**Comment by gsmet** [comment]
@evanchooly can you squash everything once you're done? Thanks!

**Comment by evanchooly** [comment]
@gsmet for sure.

**Review by geoand — APPROVED** [review]
Docs look good!

**Pull request merged**
